clients/src/main/java/org/apache/kafka/common/config/AbstractConfig.java

```diff
@@ -106,7 +107,7 @@ public AbstractConfig(ConfigDef definition, Map<?, ?> originals,  Map<String, ?>
 
         this.originals = resolveConfigVariables(configProviderProps, (Map<String, Object>) originals);
         this.values = definition.parse(this.originals);
-        this.used = Collections.synchronizedSet(new HashSet<>());
+        this.used = ConcurrentHashMap.newKeySet();
```

Comment:
Can we initialize this where the field is defined and add a comment on why it has to be concurrent?

Reply:
Done